### PR TITLE
Deprecate `enable_serverless_compute` on `databricks_sql_global_config` resource

### PR DESCRIPTION
**Pull request by nfx**
cc @jxbell

### CODE REVIEW

File: sql/resource_sql_endpoint.go [hunk 1]
```diff
@@ -194,6 +194,8 @@ func (a SQLEndpointsAPI) Delete(endpointID string) error {
 func ResourceSqlEndpoint() *schema.Resource {
 	s := common.StructToSchema(SQLEndpoint{}, func(
 		m map[string]*schema.Schema) map[string]*schema.Schema {
+		m["enable_serverless_compute"].Deprecated = "This field is intended as an internal API " +
+			"and may be removed from the Databricks Terraform provider in the future"
```
Comment (alexott) [inline sql/resource_sql_endpoint.go:198]: No, it should be done only for Global SQL config. There are a valid reasons for disabling serverless on the specific warehouses.

**Review by alexott — CHANGES_REQUESTED** [review]
we need to deprecate only global config

File: sql/resource_sql_endpoint.go (hunk 1 above)
Comment (nfx) [reply sql/resource_sql_endpoint.go]: reverted

**Review by alexott — APPROVED** [review]
lgtm!

**Pull request merged**